lib/v-click-outside.js

```diff
@@ -1,5 +1,5 @@
-const userAgent = navigator.userAgent.toLowerCase()
-const event = userAgent.match(/(iphone|ipod|ipad)/) ? 'touchstart' : 'click'
+const isTouch = (('ontouchstart' in window) || (navigator.msMaxTouchPoints > 0))
```

Comment:
Hi @dkulchenko, you mean [maxTouchPoints](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/maxTouchPoints) instead of `msMaxTouchPoints`?

Reply:
Apparently is not supported on Safari 😞. Maybe we can use this and also the user agent solution? What do you think?

Reply:
@ndelvalle Nope, msMaxTouchPoints is deliberate - as far as I can tell, IE is the only browser where ontouchstart feature detection currently doesn't work. See: https://hacks.mozilla.org/2013/04/detecting-touch-its-the-why-not-the-how/ or https://stackoverflow.com/a/13470899

What's the issue with Safari support? Seems to work fine for me on both iOS 11.1.1 Safari and macOS 10.13 Safari.

Reply:
The issue was with `maxTouchPoints` according to [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/maxTouchPoints) but we are not using it so it does not matter. This looks good 🙂 thanks for the PR.